util/formatUSDPrice.ts

```diff
@@ -0,0 +1,344 @@
+]
+
+const fiatGasPriceFormatter: FormatterRule[] = [
+  { upperBound: 0.01, formatter: '<$0.01' },
+  { upperBound: 1e6, formatter: TWO_DECIMALS_USD },
+  { upperBound: Infinity, formatter: SHORTHAND_USD_TWO_DECIMALS },
+]
+
+const fiatTokenQuantityFormatter = [{ exact: 0, formatter: '$0.00' }, ...fiatGasPriceFormatter]
+
+const portfolioBalanceFormatter: FormatterRule[] = [
+  { exact: 0, formatter: '$0.00' },
+  { upperBound: Infinity, formatter: TWO_DECIMALS_USD },
+]
+
+const ntfTokenFloorPriceFormatterTrailingZeros: FormatterRule[] = [
+  { exact: 0, formatter: '0' },
+  { upperBound: 0.001, formatter: '<0.001' },
+  { upperBound: 1, formatter: THREE_DECIMALS },
+  { upperBound: 1000, formatter: TWO_DECIMALS },
+  { upperBound: 1e15, formatter: SHORTHAND_TWO_DECIMALS },
+  { upperBound: Infinity, formatter: '>999T' },
+]
+
+const ntfTokenFloorPriceFormatter: FormatterRule[] = [
+  { exact: 0, formatter: '0' },
+  { upperBound: 0.001, formatter: '<0.001' },
+  { upperBound: 1, formatter: THREE_DECIMALS_NO_TRAILING_ZEROS },
+  { upperBound: 1000, formatter: TWO_DECIMALS_NO_TRAILING_ZEROS },
+  { upperBound: 1e15, formatter: SHORTHAND_TWO_DECIMALS_NO_TRAILING_ZEROS },
+  { upperBound: Infinity, formatter: '>999T' },
+]
+
+const ntfCollectionStatsFormatter: FormatterRule[] = [
+  { upperBound: 1000, formatter: NO_DECIMALS },
+  { upperBound: Infinity, formatter: SHORTHAND_ONE_DECIMAL },
+]
+
+export enum NumberType {
+  // used for token quantities in non-transaction contexts (e.g. portfolio balances)
+  TokenNonTx = 'token-non-tx',
+
+  // used for token quantities in transaction contexts (e.g. swap, send)
+  TokenTx = 'token-tx',
+
+  // this formatter is used for displaying swap price conversions
+  // below the input/output amounts
+  SwapPrice = 'swap-price',
+
+  // this formatter is only used for displaying the swap trade output amount
+  // in the text input boxes. Output amounts on review screen should use the above TokenTx formatter
+  SwapTradeAmount = 'swap-trade-amount',
+
+  // fiat prices in any component that belongs in the Token Details flow (except for token stats)
+  FiatTokenDetails = 'fiat-token-details',
+
+  // fiat prices everywhere except Token Details flow
+  FiatTokenPrice = 'fiat-token-price',
+
+  // fiat values for market cap, TVL, volume in the Token Details screen
+  FiatTokenStats = 'fiat-token-stats',
+
+  // fiat price of token balances
+  FiatTokenQuantity = 'fiat-token-quantity',
+
+  // fiat gas prices
+  FiatGasPrice = 'fiat-gas-price',
+
+  // portfolio balance
+  PortfolioBalance = 'portfolio-balance',
+
+  // nft floor price denominated in a token (e.g, ETH)
+  NFTTokenFloorPrice = 'nft-token-floor-price',
+
+  // nft collection stats like number of items, holder, and sales
+  NFTCollectionStats = 'nft-collection-stats',
+
+  // nft floor price with trailing zeros
+  NFTTokenFloorPriceTrailingZeros = 'nft-token-floor-price-trailing-zeros',
+}
+
+const TYPE_TO_FORMATTER_RULES = {
```

Comment:
@nenadV91 just curious, where did you get this logic from?  You did it, or is C&P from somewhere?

Reply:
Its from Uniswap, the package didn't work on Friday so I created this file. They have a pending PR to fix the issue but its not merged yet https://github.com/Uniswap/conedison/pull/18